### DISCUSSION TITLE
BirdNET-Analyzer and new models online!

### DISCUSSION THREAD

**Discussion by nilspupils** [mcguirepr89/BirdNET-Pi#184] [General]
Hi everybody, 
just wanted to let you know that Stefan Kahl @kahst released BirdNET-Analyzer with the new models. A big big "thank you" for making this awesome piece of software availeable to the public! 
As it seems @mcguirepr89, @DD4WH and @CaiusX were also involved in this! I would love to know some more details about it. 
Very exciting!
Check it out here: 
https://github.com/kahst/BirdNET-Analyzer

Best, 
Nils

Edit: 
And obviously (but totally missed by me) the one and only @mcguirepr89 is already on it!
https://github.com/mcguirepr89/BirdNET-Analyzer-Pi

Edit2: Please consider supporting the development of BirdNET and BirdNET-Pi by sponsoring or donating!

**Comment by mcguirepr89** [comment]
Hi, @nilspupils and everyone!

BirdNET-Pi on the new model is still very much in development, though if you felt like taking a peek at what it does at this point, you're more than welcome.

The installation one-liner is:
```
curl -s https://raw.githubusercontent.com/mcguirepr89/BirdNET-Analyzer-Pi/main/install_birdnetpi.sh | bash
```

At this point, the core services are there:
- Recording service
- Analysis service
- Extraction
- Streamlit
- (a new) Weather service
- and Caddy to view files during testing

There is an `uninstall.sh` script meant to be an easy way to uninstall things during testing to test the installer.

There are a few `bash` `alias`es that are included -- you can view those in the file `bash_aliases`. These are meant to make development in the terminal a bit easier for folks.

I certainly am not requesting folks test the installation yet, but I am also not going to stop anyone who is interested in trying it out at this stage.

If you do decide to play around with it, let me know what you think. Be aware that the new model has far fewer European species than the current BirdNET-Pi 6K model.

Happy birding!

**Reply by kobius77** [reply]
> Be aware that the new model has far fewer European species than the current BirdNET-Pi 6K model.

might the text in the settings of birdnet-pi be misleading then?

`Currently it only supports about 3,500 species worldwide, meaning for some regions (North America, Europe, Australia) it will usually outperform the BirdNET-Lite model, but for other regions it will be worse.`

had it activated the first night after installation as i interpreted the sentence as suggesting this model being better ('outperforming') for Europe - but just got some detections that definitly aren't common round here..

**Reply by ehpersonal38** [reply]
Patrick's comment is a year old, that was model v2.1 which had about 1000 less species than the current model has (the model has a total of 3300). That being said I don't know the exact species coverage for the new models, that text was only added after anecdotal reports from several of our European users that the new model was working well there. But if it's not for you, feel free to change back to the old model until the V2.4 6K model gets released.

**Comment by nilspupils** [comment]
@mcguirepr89:  I totally missed that you are already working on it. That is great news! How does it compare to the old model?

**Reply by mcguirepr89** [reply]
> How does it compare to the old model?

The new model and its toolset are quite impressive:
- The model has a ton of meta data that the `analyze.py` script uses to make the results more accurate -- the analysis script accepts an argument ("`--sf_thresh`") that adjusts the "`Minimum species occurrence frequency threshold for location filter`." This seems tantamount to post-processing with eBird grid data, as it essentially decides whether a species should be considered for the classification based on its recently known presence.
- The `segments.py` tool is able to fully replace the `extract_new_birdsounds.sh` script that I put together for BirdNET-Pi. This amazing little tool can be run after the analysis to extract the detections just like they are handled on BirdNET-Pi (though `extract_new_birdsounds.sh` accomplishes this using `sox` and `bash` where `segments.py` is just python). Its only "shortcoming" compared to `extract_new_birdsounds.sh` is that you are not able to set the audio format, so everything stays in the original recording format (WAVE for now).
- There is a tool called "`species.py`" that is able to generate custom species lists by simply passing it geo-coordinates. This can be helpful when you want a custom species list and do not wish to provide Latitude and Longitude during analysis. BirdNET-Pi will still use Lat and Lon, but users will be able to use the `species.py` tool to generate local species lists should they find a need for that.
- There are a "`server.py`" and "`client.py`" that basically break up the analysis the way we chopped it up for BirdNET-Pi. "`server.py`" will build the model and load the labels and expose the analysis over a port, then "`client.py`" can send the server the audio for analysis. The server can be configured to store analyses it performs while also sending JSON formatted responses with a user-defined number of prediction results returned, allowing for multiple species predictions per audio sample. (Development on this new repo currently just uses the `analyze.py` script, but will eventually migrate over to using the `server.py` and `client.py` as in the current BirdNET-Pi architecture.)
- Lastly, there is a tool with which I know not what to do! I'll need to quote the repo on this, since I don't know what it means just yet, but the "`embeddings.py`" script can be used "to extract feature embeddings instead of class predictions. Result file will contain timestamps and lists of float values representing the embedding for a particular 3-second segment. Embeddings can be used for clustering or similarity analysis." So that's cool . . . :neutral_face:
- Currently, the new model is about half the size of the 6K model we use for BirdNET-Pi, but it will continue to grow and will eventually be 6K in size, but much more powerful thanks to the metadata and the above-mentioned amazing toolset. This means European users interested in rare species might find the new model does not yet meet their needs, but I'm confident that it soon will.

As of now, I am not allowed to accept any sponsorship donations to aid in developing BirdNET-Pi around the new model. I am still allowed to accept sponsorship and donations for BirdNET-Pi built on the 6K model, however. I really hope this will change in the future so that BirdNET-Pi can continue to develop alongside the current model without hitting the same resource constraints with which I have become increasingly familiar. Cross your fingers that HaikuBox eases up on the restrictions set forth in the license to allow folks who wish to support BirdNET-Pi the freedom to do so.

My best!

PS I sent you an email to which I will kindly ask you reply :)

**Reply by nilspupils** [reply]
@mcguirepr89 : Thanks for the details! I will check it out when i have the time. Email sent!

**Comment by dbrianhendrix** [comment]
I saw that several days ago. And it's incredibly faster than previous method. I have two Audiomoths, one in front yard and one in back yard, that record the morning bird chorus from 5:30am to 8:00am CST. The recordings are in 60 second segments. I review them with Audacity, and any new "birds" that I "see" in the spectrograms, I export as a WAV file. I have a "Quick Action" (macOS) created that uses a bash script I wrote so that I just have to right click on the WAV file, select "BirdNET-AnalyzeWAV", and it starts a Docker implementation of BirdNET-Analyze, analyzes the file, and then outputs a CSV file. The script then takes that CSV file and opens it for me in Numbers which I can manipulate as I see fit like any other Numbers / Excel file. Using this process with previous BirdNET could take 20-30 seconds where now I'm seeing 3-5 second analysis time!

<img width="867" alt="Screen Shot 2022-04-29 at 8 24 32 AM" src="https://user-images.githubusercontent.com/52499277/165953223-d7cabe1d-c81c-4fde-a2ee-c154e455fd4e.png">

<img width="736" alt="Screen Shot 2022-04-29 at 8 31 04 AM" src="https://user-images.githubusercontent.com/52499277/165954341-3d694147-b428-4c23-91e1-9f73df936eeb.png">

<img width="1137" alt="Screen Shot 2022-04-29 at 8 33 14 AM" src="https://user-images.githubusercontent.com/52499277/165954605-c067cc31-d2cb-455c-a168-05975229afe0.png">

**Reply by AvnerCohen** [reply]
@mcguirepr89 Is there an easy to test run the new https://github.com/mcguirepr89/BirdNET-Analyzer-Pi alongside an existing BirdNet-Pi setup?

I am trying to see if results accuracy will improve with the new model.

**Reply by mcguirepr89** [reply]
I haven't tested them on the same system. Give me a few minutes to look over what might conflict and I'll let you know what I think.

**Reply by mcguirepr89** [reply]
So the main things that will be in conflict are the services that get installed -- they bear the same names and would end up overwriting each other.

To fix this and be able to install BirdNET-Analyzer-Pi alongside BirdNET-Pi, you _should_ be able to simply change the service names in the [installation script](https://github.com/mcguirepr89/BirdNET-Analyzer-Pi/blob/main/install_birdnetpi.sh). So changing lines 41, 55 and 56, replacing "`birdnet_analysis.service`" with something like "`birdnet_analysis_analyzer.service`". You'll do that also with the "`birdnet_recording.service`."

You'll need to change line 124 in the installation script so that it reads:
```
ExecStart=$my_dir/birdnet/bin/streamlit run $my_dir/plotly_streamlit.py --server.address localhost --server.baseUrlPath "/stats2" --server.port 8675
```
instead of
```
ExecStart=$my_dir/birdnet/bin/streamlit run $my_dir/plotly_streamlit.py --server.address localhost --server.baseUrlPath "/stats"
```
(notice the "baseUrlPath" is changed to "`/stats2`" and a new port is specified)

The other thing that will be competing is your Caddyfile.
The Caddyfile for BirdNET-Pi will look something like this:
```
http:// {
	root * /home/birdie/BirdSongs/Extracted
	file_server browse
	handle /By_Date/* {
		file_server browse
	}
	handle /Charts/* {
		file_server browse
	}
	reverse_proxy /stream localhost:8000
	php_fastcgi unix//run/php/php7.4-fpm.sock
	reverse_proxy /log* localhost:8080
	reverse_proxy /stats* localhost:8501
	reverse_proxy /terminal* localhost:8888
}
```

You would manually adjust it to look like this to support both installations:
```
http:// {
	root * /home/birdie/BirdSongs/Extracted
	file_server browse
        handle_path /BirdNET-Analyzer-Pi* {
                root * /home/birdie/BirdNET-Analyzer-Pi
                file_server browse
        }
	handle /By_Date/* {
		file_server browse
	}
	handle /Charts/* {
		file_server browse
	}
	reverse_proxy /stream localhost:8000
	php_fastcgi unix//run/php/php7.4-fpm.sock
	reverse_proxy /log* localhost:8080
	reverse_proxy /stats* localhost:8501
        reverse_proxy /stats2* localhost:8675
	reverse_proxy /terminal* localhost:8888
}
```

Note the new sections are these:
```
        handle_path /BirdNET-Analyzer-Pi* {
                root * /home/birdie/BirdNET-Analyzer-Pi
                file_server browse
        }
```
and
```
        reverse_proxy /stats2* localhost:8675
```

Be sure to change the `/home/birdie` portion to your actual username.

After changing your Caddyfile, be sure to `sudo systemctl reload caddy` to load the changes.

I'm hopeful that will let you run both systems on the same machine, though I want to warn you that this is still untested and could result in undesirable consequences/outcomes.

My very best regards,
Patrick

**Reply by ehpersonal38** [reply]
Here's some quick statistics where I took 5 random mp3 files and analyzed them through both BirdNET-Pi's model and BirdNET-Analyzer's

BirdNET-Pi: Northern Cardinal, **0.53**
BirdNET-Analyzer: Northern Cardinal, **0.74**

BirdNET-Pi: Boat-tailed Grackle, **0.48**
BirdNET-Analyzer:  Boat-tailed Grackle , **0.74**

BirdNET-Pi: Mourning Dove, **0.48**
BirdNET-Analyzer: Mourning Dove, **0.66**

BirdNET-Pi: Sandhill Crane, **0.89**
BirdNET-Analyzer: Sandhill Crane, **0.99**

BirdNET-Pi: Wild Turkey (wrong), **0.67**
BirdNET-Analyzer: Wild Turkey (wrong), **0.51**  <--- A little better at weeding out false positives

Do report back with your findings once you get everything up and running!

**Reply by mcguirepr89** [reply]
Yes, the new model hands down outperforms for all North American species

**Reply by AvnerCohen** [reply]
Thanks for the help @mcguirepr89 and @ehpersonal38 This is very encouraging for NA species, I am hoping to see something similar for this area, mainly WP birds, with some african and asian species.

That being said, I was unsuccessful with the install so far. 
I am actually not quite sure what do expect, so maybe I'll start with this:
The caddyserver config on the Analyzer repo is very different (https://github.com/mcguirepr89/BirdNET-Analyzer-Pi/blob/main/install_birdnetpi.sh#L137)
So I am wondering if I should even expect to see the full setup (including the PHP ui that provides statistics, recordings and species lists)
Or is the *Analyzer repo not yet ready for that and still require manual analysis of the recordings ?

I am able to run the `./analyze.py` manually, but I am looking for the "whole package" and more of an automatic analysis of recordings at this point.

**Reply by mcguirepr89** [reply]
The BirdNET-Analyzer-Pi repo is still very rudimentary, but does have all of "Core" analysis features:
- Recording Service
- Analysis Service
- Extraction Service
- Weather Service (new)
- Streamlit Analysis (the "Species Stats" page)

Though there is no other web interface, just the files and the Streamlit analysis. As mentioned, you'll want to manually adjust your Caddyfile using the Caddyfile in my comment as a template to get BirdNET-Analyzer-Pi data to be served alongside your BirdNET-Pi data.

It may be too much trouble to have them function in tandem :/

**Reply by AvnerCohen** [reply]
Thanks. I understand that, so will  not be looking to run the two in parallel.
I simply want to test run the new model. Things are still not working, I'll try to troubleshoot that or will start from scratch.
My understand is that running this https://github.com/mcguirepr89/BirdNET-Analyzer-Pi#setup-raspberry-pi-raspios-arm64-lite-for-testing is basically all I need to setup the full website again. Is that correct?

**Reply by mcguirepr89** [reply]
That is partially correct --  let's open an issue there to discuss this.

I'll see about running a test installation soon to see if I encounter any errors.

For the issue on the BirdNET-Analyzer-Pi repo, after installing with the `curl` command, let me know what you see when you go to `http://<THE_RASPBERRY_PI_HOSTNAME>.local` and `http://<THE_RASPBERRY_PI_HOSTNAME>.local/stats`

>My understand is that running this https://github.com/mcguirepr89/BirdNET-Analyzer-Pi#setup-raspberry-pi-raspios-arm64-lite-for-testing is basically all I need to setup the full website again. Is that correct?

Again, there are only the core services for this repo and a minimal file serving web browser -- the Streamlit data visualization is available as well, but no other web interface, so no this will not setup the full website again. It will, however, record, analyze and extract the data while offering Streamlit data visualization at `http://<THE_RASPBERRY_PI_HOSTNAME>.local/stats` after install.

We'll troubleshoot further on that repo if you'd like ~~

My best,
Patrick

**Comment by tomwhite101** [comment]
Great work Patrick. Here's hoping that the various restrictions are lifted and the new model can cover the rest of the world too!

**Comment by jds3d** [comment]
Is the BirdNET-Analyzer a replacement system for BirdNET-Pi but with the updated model?  I tried installing it on the same system using  "curl -s https://raw.githubusercontent.com/mcguirepr89/BirdNET-Analyzer-Pi/main/install_birdnetpi.sh | bash" and then went to the http://birdnetpi1.local/ website to check it out but all I see now is the file system. I'd really love to try out the new model, as I get a ton of false positives that Merlin can easily identify.  For example, today it heard a Blackpoll Warbler that was correctly identified as a Northern Cardinal by Merlin.  

Or alternatively, is it only a command line analyzer at this point in time and I should wait for you to finish the work?

**Comment by mcguirepr89** [comment]
@nilspupils !!!!

## Login:
![Screenshot from 2022-08-22 18-18-01](https://user-images.githubusercontent.com/60325264/186028478-25c5c261-903a-4d45-8c89-bde59fca91ce.png)

## Detections:
![Screenshot from 2022-08-22 18-19-43](https://user-images.githubusercontent.com/60325264/186028864-48302361-2359-4e74-a67d-4ba48e44de30.png)

## User profile management (username, password, Google Auth 2FA):
![Screenshot from 2022-08-22 18-20-09](https://user-images.githubusercontent.com/60325264/186028920-4ee2a92f-5fd9-4f7a-b53c-12ca2f575f04.png)

## Manage Browser/Session Cookies:
![Screenshot from 2022-08-22 18-20-36](https://user-images.githubusercontent.com/60325264/186028969-08a446a4-2e01-4e9e-876d-6c297f28bdfe.png)

## Set user avatar image:
![Screenshot from 2022-08-22 18-21-03](https://user-images.githubusercontent.com/60325264/186029041-79bf7628-45f4-4a09-9bf0-16e7a885c728.png)

## Edit the BirdNET-Analyzer Config File (config.py):
![Screenshot from 2022-08-22 18-21-31](https://user-images.githubusercontent.com/60325264/186029094-1be85f5a-8f47-433c-a511-b8a6a5ef2102.png)
![Screenshot from 2022-08-22 18-21-48](https://user-images.githubusercontent.com/60325264/186029121-b9ef3a9a-7b8a-4702-a629-f09759798f4f.png)
![Screenshot from 2022-08-22 18-22-02](https://user-images.githubusercontent.com/60325264/186029144-009d05be-e4ba-48fe-8007-28eaaa2d8f40.png)
![Screenshot from 2022-08-22 18-22-37](https://user-images.githubusercontent.com/60325264/186029216-2be2d7bd-fffc-4748-80fc-1fa26b463ceb.png)

## API TOKENS!!!!!! Super easy to issue and set permissions:
![Screenshot from 2022-08-22 18-23-06](https://user-images.githubusercontent.com/60325264/186029274-38512673-f8bd-4398-8e29-b560fbaa61c9.png)
![Screenshot from 2022-08-22 18-23-37](https://user-images.githubusercontent.com/60325264/186029345-0998e940-491b-4a9c-b47f-98b0b6802765.png)
![Screenshot from 2022-08-22 18-23-59](https://user-images.githubusercontent.com/60325264/186029379-78158130-7910-43af-ad92-4327c463c089.png)

APIs are awesome and easy to extend!!!!
![Screenshot from 2022-08-22 18-34-27](https://user-images.githubusercontent.com/60325264/186030574-fbab5442-7133-4dd2-a3ea-6498e5833f85.png)



Coming soon

**Reply by tomwhite101** [reply]
What is going on? It looks exciting but I dont understand it!

**Reply by nilspupils** [reply]
I am even more excited and understand it even less, but let's just say something awesome is coming! Perhaps a present to BirdNET-Pi's own birthday?? 
@mcguirepr89: Love the new look! Can't wait to see it!

**Reply by Svardsten53** [reply]
Interesting! I would love to hear the plans for the development of the program.
Will the new analysis model be included and also include European species?
Will the content and page structure be the same?
Is it possible to download the new version to see what I need to do in terms of national adaptation?

**Reply by mcguirepr89** [reply]
>What is going on? It looks exciting but I don't understand it!

>I am even more excited and understand it even less, but let's just say something awesome is coming! Perhaps a present to BirdNET-Pi's own birthday??

Mystery!!!!! Intrigue!!!! BirdNET-Pi now has that Hollywood touch!! haha jk

:)

I'm rebuilding BirdNET-Pi using the [Laravel](https://laravel.com/) PHP framework. I'm hopeful that the framework will provide a consistent and reliable structure around which folks can develop with more _insight_ and _confidence_ (***insight** into the code and application design,* ***confidence** that they won't break other components when adding their own*)

There are still TONS of things to add to the new application to ensure it is able to do everything the current version of BirdNET-Pi does, but it will (should) do those things more securely, with scalability, and will empower users to bring their own new features by leveraging the application's API for interaction/administration/data gathering/data sharing/who knows what else!

> I would love to hear the plans for the development of the program.

>Will the new analysis model be included and also include European species?

1. It _is_ built using the newest model from https://github.com/kahst/BirdNET-Analyzer.git, so the model will continue to grow and will receive improvements, optimizations, and new-species updates as time passes -- pretty cool!

>Will the content and page structure be the same?

2. The content will mostly be the same, but the page structure and layout will be totally new. Laravel employs the "[Model-view-controller](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller)" architectural design, which basically means that responsibilities within the application are better isolated from one another. The "model" will handle any data (from the database, an API, or anywhere there is data we want to bring into the application, the _model_ will fetch it for us based on the instructions we provide) -- the "controller" will perform any logic on the data that the model fetches, preparing the data to be passed to the "view." This can include sorting, transforming, translating, file building, sending HTTP requests, or anything else you need to do to data before presenting it to the user. The "view" is intuitive -- it is the last step of the puzzle where the user is presented the data and often has the ability to interact with the data. Whatever the user requests is sent back to the controller, and if necessary, the controller will communicate with the model to fetch new data, update existing data, delete data, etc.

>Is it possible to download the new version to see what I need to do in terms of national adaptation?

3. Lars, you'll be happy to know that [Laravel will support you in your localization](https://laravel.com/docs/9.x/localization#main-content)! This will let us really get BirdNET-Pi setup as an international application. I will appreciate all of your continued help on this as I am able to provide these tools to you all through testing. With that said, however, there is not an easy-to-install version that can be tested just yet. I will be sure to let you know as soon as one is available for you to begin tinkering with.

**Reply by Svardsten53** [reply]
Thanks for a good and detailed explanation. I am really looking forward to contributing with the translation and will of course have some other points of view as well.  :wink:

**Reply by krummrey** [reply]
Sounds great! Will there be an upgrade path for my existing installation, or will I have to start all over with a new install?

**Reply by matthew73210** [reply]
This is great news, thanks for the effort that you are putting in developing such a project. As a boon there's a lot of development between docker / laravel so I'm sure it'll be an easier gateway of getting it running (although this isn't the main scope of your project, in fine i'd like to see it on docker as it would be available on every platform, not just rpi ish). Might even give it a go on my side when I get back from holiday.

**Reply by rangerdavid** [reply]
LOVE THIS. I work for the National Park Service, and I would like to set up a kiosk that looks like your "Detections" screenshot. However, to really make this "pop" for visitors in our visitor center (with a microphone run outside for sound detections), it would be great to have a small image of each species as well as the name - the name alone might not mean anything to our visitors. How tough would it be to tie into an AllAboutBirds thumbnail of the species for display there? We envision this running 24/7 on a display, potentially without user interaction (or very limited), just a running live list of detections. We have a National Wildlife Refuge nearby that would want the same.

**Reply by ehpersonal38** [reply]
Hi @rangerdavid! I actually considered becoming a park ranger at one point, a little envious 😄. So as for the feature you're talking about, it's currently implemented in the latest version of BirdNET-Pi (this specific thread is for a different version that isn't being worked on anymore),

You can see it here on the "Overview" page.
![chrome_yQjDu7IAB2](https://user-images.githubusercontent.com/103586016/211673087-ccf1ca37-0bae-42ab-b16a-4d9a1ee372b9.png)

Is that the feature you're looking for? I could potentially tidy it up a bit and include more than just 5 species, or enough to fill the page completely. 

But yeah would love to hear updates about this if you decide to implement this in your visitor center, I think that's really cool!

**Reply by rangerdavid** [reply]
Thanks for the update, and hey, there's always time to become a park ranger! 

Okay, I'm a little confused by the different versions, I thought there was one that was now using an updated detection algorithm, "Analyzer" or some-such. Is that now part of the main BirdNET-Pi fork/project?

And yes, the above screenshot is pretty close to what we would need. For a "Kiosk Mode" I think the interface could be 90% a scrolling list of detection in the last 24 hours (so it isn't too long) that someone with  a touchscreen could scroll down through. Also, 10% a fixed box or header on the screen where the site name and daily and annual species detection totals (like: Tulelake Wildlife Refuge Discovery Marsh: 12 unique species today, 210 species this year) would always display. And of course, no other buttons allowing the user to leave the screen without a keyboard - like, "Esc" or other key would return to a normal page. So I can lock the RPi in a cabinet below a touchscreen digital exhibit in the visitor center, and access it with a keyboard as needed. Otherwise, it just runs like an appliance 24/7.

Thoughts?

**Reply by Svardsten53** [reply]
How about implementing this request on a hidden page since it is a special request for visitors?

I myself have been on the same track and worked a bit with making an alternative start page that instead shows the spectrogram view.

**Reply by ehpersonal38** [reply]
@rangerdavid 
> Okay, I'm a little confused by the different versions, I thought there was one that was now using an updated detection algorithm, "Analyzer" or some-such. Is that now part of the main BirdNET-Pi fork/project?

There's a separate project called BirdNET-Analyzer, it's basically the main repo where the BirdNET model gets updated and released to the public. Patrick (the creator of BirdNET-Pi) started working on a new version of BirdNET-Pi using the BirdNET-Analyzer model, but he's shelved it.

Also, I've done some work on the "kiosk mode" and pushed it to the main repo, so you're able to check it out if you have a working BirdNET-Pi installation by going to Tools -> System Controls -> Update. After that's finished, you can access kiosk mode by manually going to the URL `http://birdnetpi.local/views.php?view=Kiosk`, replacing `birdnetpi.local` with your respective installation URL or IP address. 

I've done quite a bit of optimizations such as: 

- removing all the buttons users could click
- adding a "scroll back to top" button in the bottom right of the page
- having the detection list auto-refresh every minute
- changing the detection time to a more friendly relative time, so "17:39:24" may be converted into "1 minute ago," for example. 

If you're have set your Flickr API key in Tools->Settings you'll also get bird images next to each detection which are clickable to open a larger view of the image.

![chrome_47BJq4h6ea](https://user-images.githubusercontent.com/103586016/211935033-074802b9-0061-4646-8581-be02861f74bb.png)

So, check that out if you're able to, and let me know any feedback you have. 😄

**Reply by rangerdavid** [reply]
Boy-howdy, that was fast!!! I'm going to update and check this out. If/when we have it installed, I'll post photos!

MANY THANKS!!!!

**Reply by jflyons** [reply]
Love to see the photos of your installation, and even better, visit Tulelake NWR and see it in action :) (We're nearly neighbors!)

**Comment by ehpersonal38** [comment]
Hi,

Over the past few days I've done quite a bit of work in implementing the new model into BirdNET-Pi. I realize that there has been a lot of hesitancy surrounding its implementation, mainly in regard to its poor coverage of worldwide species, so for now the old BirdNET-Lite 6K model will still be the default, with the new BirdNET-Analyzer 3k model being available on an **explicit opt-in basis** in the BirdNET-Pi settings. Hopefully this means that for users in regions where the new model excels, it will be available for use, and for everyone else, nothing will change. 

But, I need testers! So if you're willing to try out the new model, this is how you can:

**EDIT: The new model is now part of the main branch, meaning as long as you're updated via Tools -> System Controls -> Update, you will be able to access the new model from Tools -> Settings.**

<details>
  <summary>outdated steps</summary>
1. On your terminal, run the following update command: `bash ~/BirdNET-Pi/scripts/update_birdnet.sh -b new_model`. Wait for a minute or two while it runs.
2. On your BirdNET-Pi web interface, navigate to Tools -> Settings -> Model, and select the new model option under "Select a ModeL" ([the second option](https://i.imgur.com/p4Ps95w.png)).
3. Click "Update Settings" at the bottom, wait for it to do its thing, then check "View Log" at the top of the page to make sure nothing has catastrophically broken.
4. Please then navigate back to Tools -> Settings -> Model, and verify that your model settings have in fact saved and not been cleared or reset back to the defaults or any other bug.


**If you encounter any errors or failures or successes along the way please let me know in the discussion dedicated to this new model branch: https://github.com/mcguirepr89/BirdNET-Pi/discussions/699**

Hopefully with all your help I can get this pushed to the main branch of this repo in the coming days.
🐤 
</details>

**Reply by DD4WH** [reply]
@ehpersonal38 
That sounds really great !
I will try that and report back, maybe you would like to open a fresh and new discussion for testing of that particular branch ?

**Reply by ehpersonal38** [reply]
### **I've gone ahead and created a new discussion post, so all future discussions should be posted here: https://github.com/mcguirepr89/BirdNET-Pi/discussions/699**

**Comment by Punkley** [comment]
I can't help but not the new model is up to 2.3 now
https://github.com/kahst/BirdNET-Analyzer

Can we get it upgraded :)

**Reply by kahst** [reply]
We're also working on a new 6K version of the model, which is almost done training and will be released sometime next month :)

**Reply by Svardsten53** [reply]
Which of the two new models (3K and 6K) do you think would be most suitable for use in Europe? I use the new 3K in Sweden now and am very satisfied. Is the only difference that 6K will contain more species?

**Reply by kahst** [reply]
Yes, on paper, both models have the same performance for the 3K species and we "just" added more species to the list. V2.3 has a larger embedding size which lets us do a bit more stuff with embeddings and V2.4 (the upcoming 6K model) shares the same backbone and we swapped the classifier.

**Reply by ehpersonal38** [reply]
I will get this added right now! Thanks kahst for maintaining BirdNET!

**Reply by Svardsten53** [reply]
I would be very grateful if, as last time, you could change the Swedish species list so that each species name begins with a capital letter. I assume that there will be a new species list with the updated model and there the species names are misspelled due to the fact that the list is apparently based on data from eBird.
I have reported this error to eBird, but the hope that it will change is perhaps not so great.

Thank you!

**Reply by ehpersonal38** [reply]
@Svardsten53 Sure, I've just done a push which should hopefully make the changes as requested. Let me know if something doesn't work right and I'll take another look.

**Reply by Svardsten53** [reply]
Thanks! It's bed time here now and I'll try an update tomorrow. My friend Tore has already updated and I guess he will check this,

**Reply by Svardsten53** [reply]
Please see my comment here: https://github.com/mcguirepr89/BirdNET-Pi/commit/0e262ae18ef5eac7f104412e046d1e37be622892

**Comment by Punkley** [comment]
6k currently tells us we have penguins visiting our home :), ill be interested in seeing what v2.4 does (id be happy to test)

**Reply by kahst** [reply]
Yeah, V2.4 will be very different from the "old" 6K model. Yet, with more species come more false positives :)

**Comment by jds3d** [comment]
Shouldn't penguins be ruled out completely via lat/lon?

**Reply by kahst** [reply]
With V2.4 they will be, because there, we use the metadata filter as binary mask to exclude unlikely species. In the old version, we used a hybrid approach that combined audio and metadata and made a prediction based on both, so if the audio detector says "yes" with high enough confidence, it will show up as a detection independent from the metadata.

**Reply by nkwsy** [reply]
So on v2.2 & v2.3 I get terrible results. Nearly all my results are geese being called Tree Pipits or coscoroba swan's. I am located in Chicago, I am feeling like the GPS coordinates are not really working correctly. 

Below is what I see in the log. 
`13:16:56---BirdNET-Pi/birdnet/bin/python3 BirdNET-Pi/scripts/analyze.py --i BirdSongs/StreamData/2023-04-26-birdnet-RTSP_2-11:39:57.wav --o BirdSongs/StreamData/2023-04-26-birdnet-RTSP_2-11:39:57.wav.csv --lat 42.XX --lon -87.XX --week 17 --overlap 0.0 --sensitivity 1.00 --min_conf 0.6 --birdweather_id "IN_USE"`

2 things, Is there an issue w/ the GPS and species lists for v2.x? How can I have the program analyze all stream data?

**Comment by Datore7** [comment]
Hi, did an update now to latest model, but something is not working

![bild](https://user-images.githubusercontent.com/62485051/234101009-1657cbe4-c416-4017-9883-2c2cce773e5e.png)

**Reply by Datore7** [reply]
After last update, all looking good!

**Comment by timsterc** [comment]
So cool continuing to see the enthusiasm for this project - and all the continued updates!

A couple requests - from a BirdWeather perspective.

1 - When the 2.4 model is deployed, if you're able to add to this code, it would be awesome as I'm trying to make sure to keep track of which model was used for the detection.  It's already a bit out of date as the V2.3 got updated but the algorithm text being posted to BW still reflects 2.2.  

```
                                        if model == "BirdNET_GLOBAL_3K_V2.3_Model_FP16":
                                            post_algorithm = "\"algorithm\": " + "\"2p2\"" + ","
                                        else:
                                            post_algorithm = "\"algorithm\": " + "\"alpha\"" + ","


```

2 - On the BirdWeather ID Request form, instead of station name, it would be wonderful if you could update the form to ask for them to provide City/State/Country - as that's how I prefer to name the stations.   A fair bit of the time it seems like Lat/Lon wind up being their ISP, and if they don't provide their proper location, it turns into a fair bit of back/forth to get a station setup.

Thanks!
-Tim

**Reply by ehpersonal38** [reply]
Done!

**Reply by timsterc** [reply]
Lightning fast - Thank you!!!

**Comment by Punkley** [comment]
oooo 2.4 is out with 6000 birds, 

any chance of getting it pushed out :)

**Reply by ehpersonal38** [reply]
Yep, just updated it!

@Datore7 mind working your magic for the Swedish labels?

Also I guess since this has more species than the old model now I'll make it the default.

**Reply by Datore7** [reply]
Yes @ehpersonal38 , I have changed labels_sv.txt now.
[labels_nm.zip](https://github.com/mcguirepr89/BirdNET-Pi/files/11649488/labels_nm.zip)

**Reply by ehpersonal38** [reply]
Thanks @Datore7!

@DD4WH I will check with you since you were involved in this discussion a while back, do you have anything against the [new v2.4 model](https://github.com/kahst/BirdNET-Analyzer#1-model-version-update) becoming the BirdNET-Pi default?

**Reply by Datore7** [reply]
Thanks @ehpersonal38, 
I noted that out of all there are 4 other countries whose names do not have the initial capital of the local birdname, cs, no, pl and sk.
I have changed these so all txt files follow the same pattern, except the Asian and Arabic ones of course which I know nothing about...

If you think it is desirable, I will post the zip-file.

**Reply by Datore7** [reply]
@ehpersonal38  I forgot to change som lines in labels_sv.txt, it's correct now. 
[labels_nm.zip](https://github.com/mcguirepr89/BirdNET-Pi/files/11657781/labels_nm.zip)

I still have the zip with the fixed Uppercase for cs, fi, no, pl and sk, I upload that zip file if you want to use it.

**Reply by ehpersonal38** [reply]
Thanks!

I'm not sure if we should do it for other languages unless someone requests it. I think it will cause problems for people with existing installations by creating two versions of each species, the lowercase version and the capitalized version.

**Reply by Datore7** [reply]
@ehpersonal38 It is true, (I see it is mixed in the local names, some have uppercase and some lowercase initial letter. )
Anyway, just let me know and I'll fix it in later versions.

**Reply by DD4WH** [reply]
> Thanks @Datore7!
> 
> @DD4WH I will check with you since you were involved in this discussion a while back, do you have anything against the [new v2.4 model](https://github.com/kahst/BirdNET-Analyzer#1-model-version-update) becoming the BirdNET-Pi default?

First of all, thanks a lot for all your hard work in improving BirdNET-Pi and issueing updates constantly!

No, I have no objections against any new model becoming the default, as long as the oldest BirdNET-Lite model remains an option to choose in the Settings ;-).  

The model v2.4 is so new (from June 2023), that nobody really knows about its performance. But if it is set as the default, there will surely be more replies on it signalling whether it makes reasonable detections or not.

**Reply by nilspupils** [reply]
Hey! Greetings to all!
@mcguirepr89: Still in love with BirdNET-Pi, all of it exists only because of you! I hope you are doing fine!
@ehpersonal38: Thanks so much for your tireless work! You really keep everything alive. 
@DD4WH: Awesome to still have you on board. I hope you can use BirdNET-Pi for research, too?
Apart from that, i am using the new model and it works very well over here. I had 1500 detections only for yesterday in my garden setup. The ones i checked were spot on. 
Best regards, 
Nils
